client/state/reader/follows/reducer.js

```diff
@@ -20,21 +22,34 @@ import {
  * @param  {Object} action Action payload
  * @return {Array}        Updated state
  */
-export function items( state = [], action ) {
-	switch ( action.type ) {
-		case READER_FOLLOW:
-			return union( state, [ action.url ] );
-
-		case READER_UNFOLLOW:
-			return without( state, action.url );
-
-		case SERIALIZE:
-		case DESERIALIZE:
-			return [];
-	}
-
-	return state;
-}
+export const items = createReducer( {}, {
+	[ READER_FOLLOW ]: ( state, { url } ) => {
+		const urlKey = prepareComparableUrl( url );
+		return {
+			...state,
+			[ urlKey ]: { is_following: true },
+		};
+	},
+	[ READER_UNFOLLOW ]: ( state, { url } ) => {
+		const urlKey = prepareComparableUrl( url );
+		return {
+			...state,
+			[ urlKey ]: { is_following: false },
+		};
+	},
+	[ READER_FOLLOWS_RECEIVE ]: ( state, { follows } ) => {
+		const keyedNewFollows = reduce( follows, ( hash, follow ) => {
```

Comment:
I actually started out with `keyBy` and `map`, then plumped for `reduce` because otherwise we end up having to duplicate the comparable URL in the follow object (we only really want it in the key).

Is the `keyBy` version easier to understand? (Thoughts @blowery?)